### ISSUE TITLE
DefEndAlignment Cop does not autocorrect

### ISSUE THREAD

**Issue by ashmaroli** [rubocop-hq/rubocop#6219]
## Expected behavior

Expected Rubocop to correct the offense to proper style since the cop responds_to `:autocorrect`
https://github.com/rubocop-hq/rubocop/blob/9a4d88f53305cabbc79cd510ca3fcb6cdb4637df/lib/rubocop/cop/layout/def_end_alignment.rb#L64-L70

## Actual behavior

The offense remains as is at the end of session.

## Steps to reproduce the problem

```ruby
# frozen_string_literal: true

def my_method
  'foobar'
  rescue StandardError
   'foobaz'
  end
```
```bash
$ bundle exec rubocop -a test.rb
Inspecting 1 file
W

Offenses:

test.rb:7:3: W: Layout/DefEndAlignment: end at 7, 2 is not aligned with def at 3, 0.
  end
  ^^^

1 file inspected, 1 offense detected
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ bundle exec rubocop -V
0.58.2
```

**Comment by koic** [comment]
The auto-correct of this `Layout/DefEndAlignment` cop is `false` by default.
https://github.com/rubocop-hq/rubocop/blob/v0.58.2/config/default.yml#L283

It is enabled by setting to `true` in .rubocop.yml.

```yaml
Layout/DefEndAlignment:
  AutoCorrect: true
```

**Comment by ashmaroli** [comment]
Thank you @koic :+1:

**Comment by koic** [comment]
You are welcome!
